OneCommunityGlobal/HGNRest
# Aishwarya-Purchase-Consumable-Request-Form-(Backend)

Aishwarya_Purchase-Consumable-Request-Form-(Backend)

Description
This pull request adds the backend logic for the Purchase Request: Consumable form in the BM Dashboard (Phase 2).

Related PRS (if any):
This PR must be checked out to test #2018 frontend PR

Main changes explained:

Add POST route to either create a new document in buildingInventoryItem collection (if the consumablel is being ordered to the project for the first time), or update purchase record sub-document in an existing document (if the material has been ordered previously for the project). In either case, the purchaseRecord sub-document will have default values added to date, requestedBy, and status fields, in addition to the submitted form data.

How to test:
git checkout Aishwarya_Purchase-Consumable-Request-Form-(Backend)
npm install
npm run dev
check the related frontend PR for further instructions

Optional: connect to the dev db via Mongo Compass, open the buildingInventoryItem collection, and observe that documents and sub-documents are being updated correctly.

Notes
Please remember to either Approve or Request Changes. Do not just leave a comment.
Please leave a review for both the front and backend branches after you have finished testing!
